### PR TITLE
Convert MediaCard and LoadingIndicator to ES6

### PR DESCRIPTION
**Pull request by wgranger**
Staging branch URL: https://convert-small-components.pfe-preview.zooniverse.org/

Describe your changes.
I initially had plans to convert several small components to ES6 but only converted two components after realizing how many components were using the loading indicator. This should work for the two components above. The only notable difference is that the MediaCard component now applies "style" to a div directly instead of using the spread operator.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Needs a rebase.

**Comment by wgranger** [comment]
Just pushed a rebase. Trying to get this branch reviewed soon. It touches so many files that I'm having to rebase every few days.

File: app/classifier/mini-course.cjsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ React = require 'react'
 createReactClass = require 'create-react-class'
 Dialog = require 'modal-form/dialog'
 Translate = require 'react-translate-component'
-MediaCard = require '../components/media-card'
+`import MediaCard from '../components/media-card';`
```
Comment (eatyourgreens) [inline app/classifier/mini-course.cjsx]: Not a blocking comment, but backticked imports break tests, so I prefer `require().default` to import ES6 modules in coffeescript.

Comment (wgranger) [reply app/classifier/mini-course.cjsx]: Makes sense. I can take a look at all the coffescript files this touches and push a fix.

File: app/components/loading-indicator.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const LoadingIndicator = ({ children, off }) => {
+  const visibility = off ? 'hidden' : '';
```
Comment (eatyourgreens) [inline app/components/loading-indicator.jsx]: I think this should strictly be `const visibility = off ? 'hidden' : 'visible';` but it's not a big deal.

File: app/partials/sign-in-form.cjsx [hunk 3]
```diff
@@ -63,8 +63,8 @@ module.exports = createReactClass
             </a>
           </div>
 
-        else if @state.busy
-          <LoadingIndicator />
+        # else if @state.busy
```
Comment (eatyourgreens) [inline app/partials/sign-in-form.cjsx]: Should this be deleted?

Comment (wgranger) [reply app/partials/sign-in-form.cjsx]: No, I don't think it should be commented out. I think this was leftover from when I was testing things while making the conversion. I'll put it back in. Good catch.

**Pull request merged**